lighthouse-core/gather/gatherers/source-maps.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer.js');
+const Driver = require('../driver.js'); // eslint-disable-line no-unused-vars
+const LHError = require('../../lib/lh-error.js');
+
+/* global fetch */
+
+/**
+ * This function is careful not to parse the response as JSON, as it will
+ * just need to be serialized again over the protocol, and source maps can
+ * be huge.
+ *
+ * If an error occurs, the first character is '!'.
+ *
+ * @param {string[]} urls
+ * @return {Promise<string[]>}
+ */
+/* istanbul ignore next */
+async function fetchSourceMaps(urls) {
+  const responses = urls.map(async (url) => {
+    try {
+      const response = await fetch(url);
```

Comment:
should we timeout each individual fetch rather than fail the entire thing on `evaluateAsync` timeout?

Reply:
1 - one call to `evaluateAsync`, give it a big protocol timeout, give each individual fetch a reasonable timeout

2 - one call to `evaluateAsync` for each map, give reasonable protocol timeout

Which one?

Reply:
2, the less magic in `evaluateAsync` strings the better :)